Makefile

```diff
@@ -113,5 +113,39 @@ release-image-name: ## Prints the image name of the release image
 release-image-tag: ## Prints the image tag of the release image
 	@echo $(IMAGE_TAG)
 
+########################
+## Scripts and Helper ##
```

Comment:
could i get a brief on what these new make functions do? i think i got a rough sense from the PR description but i think my understanding is still fuzzy. maybe it'd be easier to ask about in person since i'm not quite sure where to start.

Reply:
We'll setup some time to hop on a call, but here's a quick blurb of the make commands. I thought this would be a nice tweak for people to figure out what changes might be coming in before running the update command, but it looks like what I was seeing was only because I hadn't committed to the branch yet. Looks like you have to be in `main` to see that, so I can remove the `local` version since it won't work as expected. 

- `_get-changes` is a helper command that the other changes commands call. It has flags that the other functions will set before calling it
- `get-last-changes-summary` gives you a file list of the files changed in the last commit. But I think I might have misunderstood the docs, I thought it was for the last merge to main, but looks to be from the last commit?
- `get-last-changes` gives a more detailed explanation of the changes compared to the one above
- `get-local-changes-summary` gives a list of the changes you have in main compared to what is currently in origin/main
- `get-local-changes` gives a more detailed explanation of the changes compared to the one above
- `update-template` runs the bash script to pull from the template repo

Output from `get-last-changes-summary`
```
 Makefile                                     |  9 +++++++++
 app/Dockerfile                               | 15 +++++++++------
 app/Makefile                                 |  2 +-
 docs/infra/vulnerability-management.md (new) | 35 +++++++++++++++++++++++++++++++++++
```

Reply:
After conversation with Loren, this is being pulled out and adding to a future PR once it has been fleshed out a bit more